### PR TITLE
Feature/dex 482 apaya test app

### PR DESCRIPTION
**Pull request by EvansPie**
DEX-482 Apaya Test App

# What this PR does

Create test app
Refactors some of the Apaya logic

# Notable decisions & other stuff

List & explain ...

# Instructions on how to test this

Let other reviewers know how they can test this.

# Before merging

_QA_

- [x] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Data Models/Apaya.swift [hunk 1]
```diff
@@ -82,4 +82,91 @@ public struct Apaya {
             )
         }
     }
+    
+    class ViewModel {
```
Comment (semirp) [inline Sources/PrimerSDK/Classes/Data Models/Apaya.swift:86]: Are we sure we want this to be in the SDK too? I believe either API will return us the `carrierName` or the DEV has to do this.

File: Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerLoadWebViewModel.swift [hunk 2]
```diff
@@ -154,6 +154,7 @@ internal class ApayaLoadWebViewModel: PrimerLoadWebViewModelProtocol {
         let clientTokenService: ClientTokenServiceProtocol = DependencyContainer.resolve()
         clientTokenService.loadCheckoutConfig { [weak self] (error) in
             if let error = error {
+                Primer.shared.delegate?.checkoutFailed?(with: error)
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerLoadWebViewModel.swift:157]: are we sure we only call this method once?

File: Sources/PrimerSDK/Classes/Data Models/Apaya.swift [hunk 3]
```diff
@@ -82,4 +82,91 @@ public struct Apaya {
...
+                  let carrier = Apaya.Carrier(mcc: mcc, mnc: mnc)
+            else { return nil }
+            
+            self.carrier = carrier
+            self.hashedIdentifier = paymentMethod.paymentInstrumentData?.hashedIdentifier
+        }
+        
+    }
+
+    enum Carrier: String, Codable {
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Data Models/Apaya.swift:104]: do we need unit tests for these?

**Review by ceriksson — COMMENTED** [review]
nice! left a question or two. Gonna QA the checkout, after that we should be good.

File: Sources/PrimerSDK/Classes/Data Models/Apaya.swift (hunk 3 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Data Models/Apaya.swift:104]: Originally this was supposed to be only in the Example app's side. However, I added it on the SDK as well cause we might need it. I can add some tests.

File: Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerLoadWebViewModel.swift (hunk 2 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerLoadWebViewModel.swift:157]: I believe so, it is not called from `navigate`, nor the `router`.

**Pull request merged**